tests/test_configurations/test_aea_config.py

```diff
@@ -164,7 +164,7 @@ def test_builder_applies_default_value_to_aea(self, mock: Mock) -> None:
         """Test AEABuilder applies default value to AEA instance when option is not specified in config."""
         configuration = self._make_configuration(NotSet)
         builder = AEABuilder()
-        builder._set_from_configuration(configuration, aea_project_path=Path("."))
+        builder.set_from_configuration(configuration, aea_project_path=Path("."))
```

Comment:
If public, this might lead to undesired side-effects. 
The implementation "adds" stuff (components/private-keys etc) to the builder, doesn't replace it.

Reply:
An easy solution is to make it private again.